modules/govuk_mysql/manifests/xtrabackup/backup.pp

```diff
@@ -0,0 +1,113 @@
+# == Class: Govuk_mysql::Xtrabackup::Backup
+#
+# Creates base and incremental backups which are streamed to an S3 bucket.
+# Based upon this MariaDB blog post: https://mariadb.com/blog/streaming-mariadb-backups-cloud
+#
+# === Parameters
+#
+# [*aws_access_key_id*]
+#   The AWS access key used to access the S3 bucket.
+#
+# [*aws_secret_access_key*]
+#   The AWS secret access key used to access the S3 bucket.
+#
+# [*aws_region*]
+#   The region where the S3 bucket lives.
+#   Default: eu-west-1
+#
+# [*s3_bucket_name*]
+#   The unique name of the S3 bucket.
+#
+# [*encryption_key*]
+#   The encryption key used to encrypt the backups.
+#
+# [*base_backup_cron_minute*]
+#   The minute(s) that the daily base backup(s) run(s).
+#   Default: 10
+#
+# [*base_backup_cron_hour*]
+#   The hour(s) that the daily base backup(s) run(s).
+#   Default: 6
+#
+# [*incremental_backup_cron_minute*]
+#   The minute(s) that the incremental backup(s) run(s).
+#   Default: */15 (every 15 minutes)
+#
+# [*enabled*]
+#   If this is true then the packages are installed and backups are triggered.
+#   Default: false
+#
+class govuk_mysql::xtrabackup::backup (
+  $aws_access_key_id = undef,
+  $aws_secret_access_key = undef,
+  $aws_region = 'eu-west-1',
+  $s3_bucket_name = undef,
+  $encryption_key = undef,
+  $base_backup_cron_minute = '10',
+  $base_backup_cron_hour = '6',
+  $incremental_backup_cron_minute = '*/15',
+  $enabled = false,
+) {
+  if $enabled {
+    include govuk_mysql::xtrabackup::packages
+
+    file {['/etc/mysql/xtrabackup', '/etc/mysql/xtrabackup/env.d']:
+      ensure => 'directory',
+      mode   => '0755',
+      owner  => 'root',
+      group  => 'root',
+    }
+
+    file { '/etc/mysql/xtrabackup/env.d/AWS_ACCESS_KEY_ID':
+      ensure  => 'present',
+      content => $aws_access_key_id,
+      mode    => '0440',
+      owner   => 'root',
+      group   => 'root',
+    }
+
+    file { '/etc/mysql/xtrabackup/env.d/AWS_SECRET_ACCESS_KEY':
+      ensure  => 'present',
+      content => $aws_secret_access_key,
+      mode    => '0440',
+      owner   => 'root',
+      group   => 'root',
+    }
+
+    $base_threshold_secs = 28 * 3600
+    $base_service_desc = 'MySQL Xtrabackup daily base push'
+
+    file { '/usr/local/bin/xtrabackup_s3_base':
+      ensure  => 'present',
+      content => template('govuk_mysql/usr/local/bin/xtrabackup_s3_base'),
+      mode    => '0755',
+    }
+
+    @@icinga::passive_check { "check_mysql_xtrabackup_daily_base_push-${::hostname}":
+        service_description => $base_service_desc,
+        freshness_threshold => $base_threshold_secs,
+        host_name           => $::fqdn,
+      }
+
+    $incremental_threshold_secs = 20 * 60
```

Comment:
Now that this threshold has been lowered, we'd need to change the default on `incremental_backup_cron_minute` so that we don't raise false alerts.

What would the impact be do you think in terms of bandwidth, disk space and recovery time if we ran the job every 5 minutes and relied on `setlock` to avoid backups from running into each other?